vendor/bam/src/json/frame.rs

```diff
@@ -147,7 +140,10 @@ impl FrameHandler<json::Frame, json::Request, json::Response> for JsonFrameHandl
 
                 debug!("Dispatching response frame {:?} to stored handler.", frame);
 
-                sender.send(frame).unwrap();
+                let response =
+                    serde_json::from_value(frame.payload).expect("should always deserialize");
```

Comment:
🤔 And what if the other party sends garbage json as the frame payload? That can easily happen and I doubt we want to panic here.

Reply:
Ahhh well spotted. (note it panic'd before this PR too)

Reply:
True, my bad then. I think I wrote the old code 😅
I think it actually was the way we have it now at some point but we changed it to accept a `Frame` for symmetric reasons. Hence also the name, Response*Frame*Source.